analyzers/src/SonarAnalyzer.CSharp/Extensions/ArgumentSyntaxExtensions.cs

```diff
@@ -34,6 +34,15 @@ internal static class ArgumentSyntaxExtensions
                 .GetArgumentsOfKnownType(knownType, semanticModel)
                 .Select(argument => semanticModel.GetSymbolInfo(argument.Expression).Symbol);
 
+        internal static IEnumerable<ISymbol> GetSymbolsDerivedFromKnownType(this SeparatedSyntaxList<ArgumentSyntax> syntaxList, KnownType knownType, SemanticModel semanticModel) =>
+            syntaxList
+                .GetArgumentsDerivedFromKnownType(knownType, semanticModel)
+                .Select(argument => semanticModel.GetSymbolInfo(argument.Expression).Symbol);
```

Comment:
This will call `semanticModel.GetSymbolInfo()` on every argument twice. The documentation does mention some caching on GetSymbolInfo(), but I took a quick look at [the Roslyn repo](https://github.com/dotnet/roslyn/blob/829c4c50d28f388dea12dd3a962f89ee72040771/src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs#L519), and I didn't see much of it. 
Either merge the two methods or just don't call the second one from the first.
Or the second one can return a collection of pairs of `(ArgumentSyntax, ISymbol)`.
WDYT?